### PR TITLE
FIx Implementation

### PR DESCRIPTION
**Pull request by mark-hunter**
The framework now cleans up the FulfillmentCache and JobCardCache.

### CODE REVIEW

File: petasos/core/src/main/java/net/fhirfactory/pegacorn/petasos/core/tasks/management/participant/watchdogs/LocalPetasosTaskCleanupWatchdog.java [hunk 1]
```diff
@@ -162,26 +198,102 @@ protected void taskContinuityWatchdog(){
...
             }
             if(unregisterTask){
-                getLogger().debug(".taskContinuityWatchdog(): Task {} is finalised, removing from shared cache... start", currentTaskId);
+                getLogger().debug(".actionableTaskCleanup(): Task {} is finalised, removing from shared cache... start", currentTaskId);
                 PetasosActionableTask unregisteredActionableTask = actionableTaskDM.removeTask(currentTaskId);
-                getLogger().debug(".taskContinuityWatchdog(): Task {} is finalised, removing from shared cache... done...");
-            }
-            if(getLogger().isDebugEnabled()){
-                getLogger().debug(".taskContinuityWatchdog(): Shared ActionableTaskCache size->{}", actionableTaskDM.getAllTaskIds().size());
+                getLogger().debug(".actionableTaskCleanup(): Task {} is finalised, removing from shared cache... done...");
```
Comment (toneill) [inline petasos/core/src/main/java/net/fhirfactory/pegacorn/petasos/core/tasks/management/participant/watchdogs/LocalPetasosTaskCleanupWatchdog.java:203]: I think this section of logging was meant to go above in the if (unregisterTask)

File: petasos/core/src/main/java/net/fhirfactory/pegacorn/petasos/core/tasks/management/participant/watchdogs/LocalPetasosTaskCleanupWatchdog.java [hunk 2]
```diff
@@ -162,26 +198,102 @@ protected void taskContinuityWatchdog(){
...
 
+    //
+    // Fulfillment Task Controller / Watchdog
+    //
+
+    protected void fulfillmentTaskCleanup(){
+        getLogger().debug(".fulfillmentTaskCleanup(): Entry");
+        List<PetasosFulfillmentTask> allTasks = fulfillmentTaskCache.getFulfillmentTaskList();
+        for(PetasosFulfillmentTask currentTask: allTasks){
+            if(getLogger().isInfoEnabled()){
```
Comment (toneill) [inline petasos/core/src/main/java/net/fhirfactory/pegacorn/petasos/core/tasks/management/participant/watchdogs/LocalPetasosTaskCleanupWatchdog.java:249]: Should be isDebugEnabled() here

**Review by toneill — COMMENTED** [review]
Minor logging changes

**Review by mark-hunter — COMMENTED** [review]
Made changes to logging...

**Review by toneill — APPROVED** [review]

**Pull request merged**